src/wrapper.ts

```diff
@@ -6,109 +6,63 @@
-import path from "path";
 import Service from "serverless/classes/Service";
-import util from "util";
 import { FunctionInfo, RuntimeType } from "./layer";
-import { es6Template } from "./templates/node-es6-template";
-import { nodeTemplate } from "./templates/node-js-template";
-import { typescriptTemplate } from "./templates/node-ts-template";
-import { pythonTemplate } from "./templates/python-template";
-import { getHandlerPath, removeDirectory } from "./util";
-import { TracingMode } from "./templates/common";
 
-export const datadogDirectory = "datadog_handlers";
+export const datadogHandlerEnvVar = "DD_LAMBDA_HANDLER";
+export const pythonHandler = "datadog_lambda.handler.handler";
+export const pythonHandlerFile = "datadog_lambda.handler.py";
+export const jsHandlerLayerPrefix = "/opt/nodejs/";
+export const jsHandler = "node_modules/datadog-lambda-js/handler.handler";
+export const jsHandlerFile = "node_modules/datadog-lambda-js/handler.js";
 
-export interface FunctionGroup {
-  funcs: {
-    info: FunctionInfo;
-    method: string;
-  }[];
-  filename: string;
-  runtime: RuntimeType;
-}
-
-export async function writeHandlers(service: Service, funcs: FunctionInfo[], tracingMode: TracingMode) {
-  await cleanupHandlers();
-  await util.promisify(fs.mkdir)(datadogDirectory);
-  const groups = getFunctionGroups(funcs);
-  const promises = groups.map(async (funcGroup) => {
-    const methods = getMethods(funcGroup);
-    const result = getWrapperText(funcGroup.runtime, funcGroup.filename, methods, tracingMode);
-    if (result === undefined) {
+export function redirectHandlers(service: Service, funcs: FunctionInfo[], addLayers: boolean) {
+  const handlerFiles: Set<string> = new Set();
+  funcs.map((func) => {
+    setEnvDatadogHandler(func);
+    const handlerInfo = getDDHandler(func.type, addLayers);
+    if (handlerInfo === undefined) {
       return;
     }
-    const text = result;
-    const filename = await writeWrapperFunction(funcGroup, text);
-    const baseMethodName = path.posix.join(datadogDirectory, funcGroup.funcs[0].info.name);
-    for (const func of funcGroup.funcs) {
-      func.info.handler.handler = `${baseMethodName}.${func.method}`;
-      if (func.info.handler.package === undefined) {
-        func.info.handler.package = {
-          exclude: [],
-          include: [],
-        };
-      }
-      if (func.info.handler.package.include === undefined) {
-        func.info.handler.package.include = [];
-      }
-      func.info.handler.package.include.push(filename);
+    const { handler, handlerFile } = handlerInfo;
+    func.handler.handler = handler;
+
+    if (func.handler.package === undefined) {
+      func.handler.package = {
+        exclude: [],
+        include: [],
+      };
+    }
+    if (func.handler.package.include === undefined) {
+      func.handler.package.include = [];
     }
-    return `${filename}`;
+    func.handler.package.include.push(handlerFile);
+    handlerFiles.add(handlerFile);
```

Comment:
handlerFile points to a file in the layer right? Not sure if it makes sense to add it to the include list, since by the time we run this code the function has already been packaged. I don't think this has any effect.